Support tasty-1.5

```
tasty-coverage-0.1.1.0$ cabal build -c 'tasty>=1.5' --allow-newer
[1 of 1] Compiling Test.Tasty.CoverageReporter

src/Test/Tasty/CoverageReporter.hs:68:76: error: [GHC-83865]
    • Couldn't match type ‘[]’ with ‘(,) (NonEmpty TestName)’
      Expected: [(NonEmpty TestName, String -> IO ())]
        Actual: [[(NonEmpty TestName, String -> IO ())]]
    • In the second argument of ‘fmap’, namely ‘acc’
      In the expression: fmap (first (NE.cons groupName)) acc
      In the expression:
        \ _ groupName acc -> fmap (first (NE.cons groupName)) acc
   |
68 |           foldGroup = \_ groupName acc -> fmap (first (NE.cons groupName)) acc
   |                                                                            ^^^
Error: cabal: Failed to build tasty-coverage-0.1.1.0.

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,9 @@
 # Revision history for tasty-test
 
+## Unreleased
+
+* Add compatibility with Tasty 1.5
+
 ## 0.1.1.0 -- 2023-09-07
 
 * Be more specific with the file suffixes which are computed from the test outcome. Previously only "PASSED" and "FAILED" were used. We now also use "EXCEPTION", "TIMEOUT" and "SKIPPED".

--- a/src/Test/Tasty/CoverageReporter.hs
+++ b/src/Test/Tasty/CoverageReporter.hs
@@ -1,4 +1,4 @@
-{-# LANGUAGE InstanceSigs, NamedFieldPuns #-}
+{-# LANGUAGE InstanceSigs, NamedFieldPuns, CPP #-}
 module Test.Tasty.CoverageReporter (coverageReporter) where
 
 import Test.Tasty
@@ -49,10 +49,19 @@ tixDir = "tix"
 testNames :: OptionSet -> TestTree -> IO ()
 testNames  os tree = forM_ (foldTestTree coverageFold os tree) $ \(s,f) -> f (fold (NE.intersperse "." s))
 
+type FoldResult = [(NonEmpty TestName, String -> IO ())]
+
+#if MIN_VERSION_tasty(1,5,0)
+groupFold :: OptionSet -> TestName -> [FoldResult] -> FoldResult
+groupFold _ groupName acc = fmap (first (NE.cons groupName)) (concat acc)
+#else
+groupFold :: OptionSet -> TestName -> FoldResult -> FoldResult
+groupFold _ groupName acc = fmap (first (NE.cons groupName)) acc
+#endif
 
 
 -- | Collect all tests and
-coverageFold :: TreeFold [(NonEmpty TestName, String -> IO ())]
+coverageFold :: TreeFold FoldResult
 coverageFold = trivialFold
        { foldSingle = \opts name test -> do
           let f n = do
@@ -65,7 +74,7 @@ coverageFold = trivialFold
                 putStrLn ("Wrote coverage file: " <> filepath)
           pure (NE.singleton name, f),
           -- Append the name of the testgroup to the list of TestNames
-          foldGroup = \_ groupName acc -> fmap (first (NE.cons groupName)) acc
+          foldGroup = groupFold
         }
 
 tixFilePath :: TestName -> Result -> FilePath

--- a/tasty-coverage.cabal
+++ b/tasty-coverage.cabal
@@ -26,30 +26,34 @@ source-repository head
     type:     git
     location: git@github.com:BinderDavid/tasty-coverage.git
 
+common deps
+    build-depends:
+        base >=4.16 && <4.20,
+        tasty ^>=1.4
+        
 library
+    import:           deps
     exposed-modules:  Test.Tasty.CoverageReporter
     hs-source-dirs:   src
     default-language: Haskell2010
     ghc-options:      -Wall
     build-depends:
-        base >=4.16 && <4.20,
         filepath >=1.3.8 && <=1.5,
-        tasty ^>=1.4,
         hpc >=0.6 && <0.8,
         
 
 executable tasty-coverage-example
+    import:           deps
     main-is:          Main.hs
     hs-source-dirs:   example
     default-language: Haskell2010
     ghc-options:      -Wall -fhpc
     build-depends:
-        base >=4.16.0.0 && <4.20,
-        tasty ^>=1.4,
         tasty-hunit ^>=0.10,
         tasty-coverage
 
 test-suite tasty-coverage-test
+    import:             deps
     type:               exitcode-stdio-1.0
     main-is:            Main.hs
     build-tool-depends: tasty-coverage:tasty-coverage-example
@@ -57,9 +61,7 @@ test-suite tasty-coverage-test
     default-language:   Haskell2010
     ghc-options:        -Wall
     build-depends:
-        base >=4.16.0.0 && <4.20,
         filepath >=1.3.8 && <=1.5,
-        tasty ^>=1.4,
         tasty-golden ^>=2.3,
         process ^>=1.6,
         

--- a/tasty-coverage.cabal
+++ b/tasty-coverage.cabal
@@ -29,7 +29,7 @@ source-repository head
 common deps
     build-depends:
         base >=4.16 && <4.20,
-        tasty ^>=1.4
+        tasty >= 1.4 && < 1.6
         
 library
     import:           deps

